packages/ui-components/style/windowedlist.css

```diff
@@ -4,17 +4,58 @@
  */
 
 .jp-WindowedPanel-outer {
+  height: 100%;
   position: relative;
-  overflow-y: auto;
+  overflow: auto;
+  transform: none;
+}
+
+.jp-WindowedPanel.jp-mod-virtual-scrollbar > .jp-WindowedPanel-outer {
+  margin-left: 25px;
 }
 
 .jp-WindowedPanel-inner {
   position: relative;
 }
 
-.jp-WindowedPanel-window {
+.jp-WindowedPanel-viewport {
   position: absolute;
   left: 0;
   right: 0;
   overflow: visible;
 }
+
+.jp-WindowedPanel-scrollbar {
+  display: none;
+}
+
+.jp-WindowedPanel.jp-mod-virtual-scrollbar > .jp-WindowedPanel-scrollbar {
+  background-color: inherit;
+  border-left: 1px solid var(--jp-layout-color3);
+  display: block;
+  position: fixed;
+  overflow-y: auto;
+  top: 0;
+  bottom: 0;
+  right: 25px;
+  min-width: 35px;
+  z-index: 1;
+}
+
+ol.jp-WindowedPanel-scrollbar-content {
+  background-color: var(--jp-layout-color1);
+  list-style-type: none;
+  margin: 0;
+  padding: 0;
+}
+
+ol.jp-WindowedPanel-scrollbar-content > li {
+  border-bottom: 1px solid var(--jp-layout-color3);
+  padding: 2px;
+  text-align: left;
+}
+
+ol.jp-WindowedPanel-scrollbar-content > li:hover {
+  cursor: pointer;
+  background-color: var(--jp-layout-color2);
+}
```

Comment:
It is uncommon to use HTML tag for styling. Should we stick to class based styling only?

Reply:
I think this is related to your suggestion above. If the API contract is for an `<li>` element, we could leave this as is. So I suppose: how strongly do you feel about the above `<ol>` and `<li>` conversation?

Reply:
A different point: if we were to migrate to a custom webcomponent in the future (see the recent toolbar PR) not having the `ol` and `li` in style files would be somewhat easier to make this change. I somewhat prefer to exclusively use class names if possible.

Performance-wise there are arguments both ways (e.g. `li:hover` will initially match on a lot of things but will be quickly invalidated via parent, but then `.jp-WindowedPanel-scrollbar-content > li` avoids adding a class name so may be tiny bit faster to populate).

Reply:
I removed the tag names from the CSS and updated the `IRenderer` interface to not require specific tags.